scipio/src/executor.rs

```diff
@@ -322,28 +352,99 @@ impl LocalExecutor {
+    /// fine-grained error handling consider initializing those entities manually.
+    ///
+    /// You must call wait_on_executors() to wait for all executors spawned through spawn_new
+    /// to return
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use scipio::LocalExecutor;
+    ///
+    /// // executor is a single thread, but not bound to any particular CPU.
+    /// LocalExecutor::spawn_new("myname", None, async move {
+    ///     println!("hello");
+    /// });
+    ///
+    /// LocalExecutor::wait_on_executors();
+    /// ```
+    pub fn spawn_new<F, T>(name: &'static str, binding: Option<usize>, fut: F)
+    where
+        F: Send + 'static,
+        F: Future<Output = T>,
+        T: Send + 'static,
+    {
+        let id = EXECUTOR_ID.fetch_add(1, Ordering::Relaxed);
+
+        let mut executors = SPAWNED_EXECUTORS.lock().unwrap();
+        (*executors).push(
+            Builder::new()
+                .name(format!("{}-{}", name, id).to_string())
+                .spawn(move || {
+                    let mut le = LocalExecutor {
+                        queues: ExecutorQueues::new(),
+                        parker: parking::Parker::new(),
+                        binding,
+                        id,
+                    };
+                    le.init().unwrap();
+                    le.run(async move {
+                        let task = Task::local(async move {
+                            fut.await;
+                        });
+                        task.await;
+                    })
+                })
+                .unwrap(),
+        )
+    }
 
-        let io_requirements = IoRequirements::new(Latency::NotImportant, 0);
-        le.queues.borrow_mut().available_executors.insert(
-            0,
-            TaskQueue::new("default", 1000, io_requirements, move || {
-                let mut queues = queues.borrow_mut();
-                queues.maybe_activate(index);
-            }),
-        );
-        Ok(le)
+    /// Wait for all executors called through spawn_new to return
+    ///
+    /// # Examples
+    /// use scipio::LocalExecutor;
+    ///
+    /// // executor is a single thread, but not bound to any particular CPU.
+    /// LocalExecutor::spawn_new("myname", None, async move {
+    ///     println!("hello");
+    /// });
+    ///
+    /// LocalExecutor::wait_on_executors();
+    pub fn wait_on_executors() {
+        let mut executors = SPAWNED_EXECUTORS.lock().unwrap();
+        for ex in std::mem::replace(&mut *executors, Vec::new()) {
+            ex.join().unwrap();
+        }
+    }
+
+    /// Returns a unique identifier for this Executor.
+    ///
+    /// # Examples
+    /// ```
+    /// use scipio::LocalExecutor;
+    ///
+    /// let local_ex = LocalExecutor::new(None).expect("failed to create local executor");
+    /// println!("My ID: {}", local_ex.id());
+    /// ```
+    pub fn id(&self) -> usize {
+        self.id
```

Comment:
Since this id is public, it would make sense to use it to decorate panic messages. Probably not the scope of this PR but nice to keep in mind.

Reply:
:+1: